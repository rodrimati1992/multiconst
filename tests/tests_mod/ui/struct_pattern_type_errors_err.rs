use core::ops::Range;

use multiconst::{FieldType, Usize};

multiconst::multiconst!{
    const Range{start: A:u16, ..}: Range<u8> = 3..5;
}

struct Tupled(u8);

impl FieldType<Usize<0>> for Tupled {
    type Type = u8;
}

multiconst::multiconst!{
    const Tupled(B: u32): Tupled = Tupled(10);
}


fn main(){}