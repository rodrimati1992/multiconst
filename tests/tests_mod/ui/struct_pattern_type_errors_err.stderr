error[E0308]: mismatched types
 --> tests/tests_mod/ui/struct_pattern_type_errors_err.rs:6:24
  |
6 |     const Range{start: A:u16, ..}: Range<u8> = 3..5;
  |                        ^ expected `u16`, found `u8`
  |
  = note: this error originates in the macro `$crate::__::__priv_multiconst_proc_macro` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/tests_mod/ui/struct_pattern_type_errors_err.rs:16:18
   |
16 |     const Tupled(B: u32): Tupled = Tupled(10);
   |                  ^ expected `u32`, found `u8`
   |
   = note: this error originates in the macro `$crate::__::__priv_multiconst_proc_macro` (in Nightly builds, run with -Z macro-backtrace for more info)
